src/main/java/com/vmware/vipclient/i18n/messages/service/ProductService.java

```diff
@@ -72,24 +72,41 @@ public ProductService(BaseDTO dto) {
         return components;
     }
 
+    public Set<Locale> getSupportedLocales() {
+        return getSupportedLocales(true);
+    }
+
     /**
-     * Retrieves the list of locales of a product. It recursively applies data source fallback mechanism in case of failure.
+     * Retrieves the list of locales of a product.
      *
+     * @param withCacheRefresh If true, it recursively applies data source fallback mechanism in case of failure.
      * @return list of locales of the product specified in the dto object, or an empty list in case of failure to retrieve from any data source.
      */
-    public Set<Locale> getSupportedLocales() {
-        return langTagtoLocaleSet(getSupportedLanguageTags());
+    public Set<Locale> getSupportedLocales(boolean withCacheRefresh) {
+        return langTagtoLocaleSet(getSupportedLanguageTags(withCacheRefresh));
```

Comment:
It's not good to use this method everywhere. We should use only one type - set or list - internally.

Reply:
They are both Set type. List was not used. This method converts from Set<String> language tags to Set<Locale> locales

Reply:
See the updated code. I have moved this utility method to LocaleUtilityand used it for the call to LocaleUtility.pickupLocaleFromList only.